projects/spectator/src/lib/base/dom-spectator.ts

```diff
@@ -97,7 +97,8 @@ export abstract class DomSpectator<I> extends BaseSpectator {
   public setInput<K extends keyof I>(input: K, inputValue: I[K]): void;
   public setInput(input: any, value?: any): void {
     setProps(this.instance, input, value, false);
-    this.debugElement.injector.get(ChangeDetectorRef).detectChanges();
+    setMarkForCheck(this.debugElement);
```

Comment:
Invoking markForCheck when setting an input doesn't make sense to me. Why should we mark the whole tree when changing an input?

Reply:
We mark node for check need `detectChanges` and we run root `detectChanges` from `fixture`.
In this version, the all default hooks starts automatic in three if it need.
In previous version, the caller `detectChanges` running DC only on focused node.